src/Storages/StorageS3Settings.h

```diff
@@ -21,58 +21,96 @@ namespace DB
 {
 
 struct Settings;
+class NamedCollection;
 
 struct S3Settings
 {
     struct RequestSettings
     {
-        size_t max_single_read_retries = 0;
-        size_t min_upload_part_size = 0;
-        size_t max_upload_part_size = 0;
-        size_t upload_part_size_multiply_factor = 0;
-        size_t upload_part_size_multiply_parts_count_threshold = 0;
-        size_t max_part_number = 0;
-        size_t max_single_part_upload_size = 0;
-        size_t max_single_operation_copy_size = 0;
-        size_t max_connections = 0;
+        struct PartUploadSettings
+        {
+            size_t min_upload_part_size = 16 * 1024 * 1024;
```

Comment:
The default values are supposed to stay in-sync with the defaults in Core/Settings.h, right? Maybe it's possible to omit the defaults here somehow?